.github/workflows/pytest.yml

```diff
@@ -87,13 +87,18 @@ jobs:
             run_type=$(echo "${{ inputs.upstream-details }}" | jq --arg repo "$repo" '".$repo"."run-type"')
             run_id=$(echo "${{ inputs.upstream-details }}" | jq --arg repo "$repo" '".$repo"."run-id"')
             # Amalgam and howso-engine downloads need to be placed in their embedded location
+            IFS='-' read -r plat arch <<< "${{ inputs.amalgam-plat-arch }}"
             if [[ "$repo" == "amalgam" ]]; then
-              gh $run_type download -D amalgam/lib/$plat/amd64 -R "howsoai/$repo" -p "*${{ inputs.amalgam-plat-arch }}*" "$run_id"
+              echo "Downloading and extracting Amalgam binaries for $plat/$arch..."
+              gh $run_type download -D amalgam/lib/$plat/$arch -R "howsoai/$repo" -p "*${{ inputs.amalgam-plat-arch }}*" "$run_id"
               # Extract binaries
-              cd amalgam/lib
+              cd amalgam/lib/$plat/$arch && if [ ! -f *.tar.gz ]; then mv */*.tar.gz ./; fi && tar -xvzf *.tar.gz
+              cd ../..
               find . -type d -name lib -exec sh -c 'mv {}/* "$(dirname {})"' \;
```

Comment:
Is this cmd still needed after you updates above? I would think if the mv and untar work above right, no need to do another move?

Reply:
This is needed because the binaries come packaged in a `lib` directory and I need them sitting at the `$arch` directory (instead of `$arch/lib`).

Reply:
Simplified this a bit